app/src/main/java/vn/sunasterisk/realfootball/ui/detailmatch/DetailMatchFragment.kt

```diff
@@ -1,15 +1,53 @@
 package vn.sunasterisk.realfootball.ui.detailmatch
 
 import android.os.Bundle
-import android.view.LayoutInflater
-import android.view.ViewGroup
-import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import org.koin.android.viewmodel.ext.android.viewModel
 import vn.sunasterisk.realfootball.R
+import vn.sunasterisk.realfootball.base.ViewModelBaseFragment
+import vn.sunasterisk.realfootball.databinding.FragmentDetailMatchBinding
 
-class DetailMatchFragment : Fragment() {
-    override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ) =
-        inflater.inflate(R.layout.fragment_detail_match, container, false)
+class DetailMatchFragment : ViewModelBaseFragment<DetailViewModel, FragmentDetailMatchBinding>() {
+
+    override val viewModel: DetailViewModel by viewModel()
+
+    override val contentViewId get() = R.layout.fragment_detail_match
+
+    val idEvent by lazy {
+        arguments?.let {
+            DetailMatchFragmentArgs.fromBundle(it).idEvent
+        }
+    }
+    val idHomeTeam by lazy {
+        arguments?.let {
+            DetailMatchFragmentArgs.fromBundle(it).idHome
+        }
+    }
+    val idAwayTeam by lazy {
+        arguments?.let {
+            DetailMatchFragmentArgs.fromBundle(it).idAway
+        }
+    }
+
+    override fun initializeView(savedInstanceState: Bundle?) {
+    }
+
+    override fun initializeComponents() {
+        viewDataBinding.viewModel = viewModel
+        viewModel.initData(idEvent!!, idHomeTeam!!, idAwayTeam!!)
+
+        viewModel.match.observe(viewLifecycleOwner, Observer {
+            handlerError(it)
+        })
+
+        viewModel.homeTeam.observe(viewLifecycleOwner, Observer { team ->
+            handlerError(team)
+            team.result?.let { viewModel.updateHomeUrl(it) }
+        })
+
+        viewModel.awayTeam.observe(viewLifecycleOwner, Observer { team ->
+            handlerError(team)
+            team.result?.let { viewModel.updateAwayUrl(it) }
+        })
+    }
```

Comment:
Dùng apply{} nha, có cách nào tốt hơn cho 3 toán tử !! kia không ông

Reply:
Thế theo ô khi nào cái kia có thể null được

Reply:
T đang hỏi xem có cách nào xử lý được không. Còn khi nào null ông tự check chứ

Reply:
thì t ms hỏi ô xem ô bt khi nào null. !! là khi mình bt n k null r.